batchlib/analysis/background_extraction.py

```diff
@@ -120,53 +122,76 @@ def get_bg_stats(self, bg_values):
         with open_file(self.table_out_path, 'a') as f:
             self.write_table(f, table_key, column_names, table)
 
     def run(self, input_files, out_files):
-        # first, get plate wide and per-well background statistics
+        # first, get per image background statistics
         logger.info(f'{self.name}: computing background statistics')
-        bg_dict = {file: self.get_bg_segment(file, device='cpu') for file in input_files}
-        bg_per_image_stats = {in_file_to_image_name(file): self.get_bg_stats(bg_segments)
-                              for file, bg_segments in bg_dict.items()}
+
+        bg_dict = {file_name: self.get_bg_segment(file_name, device='cpu') for file_name in input_files}
+
+        # separate the bg dict into actual bg dict and background fractions
+        bg_fractions_per_image = {in_file_to_image_name(k): v[1] for k, v in bg_dict.items()}
+        bg_dict = {k: v[0] for k, v in bg_dict.items()}
+
+        bg_per_image_stats = {in_file_to_image_name(file_name): self.get_bg_stats(bg_segments)
+                              for file_name, bg_segments in bg_dict.items()}
 
         # ignore image outliers in per_well and per_plate backgrounds
         outlier_dict = _load_image_outliers(self.name, self.table_out_path, self.image_outlier_table, input_files)
+
         bg_per_well_dict = defaultdict(list)
+        bg_fractions_per_well = defaultdict(list)
+
         wells = set()
-        for file, bg_segment in bg_dict.items():
-            well = image_name_to_well_name(os.path.basename(file))
+        for file_name, bg_segment in bg_dict.items():
+            image_name = in_file_to_image_name(file_name)
+            well = image_name_to_well_name(image_name)
             wells.add(well)
-            image_name = os.path.splitext(os.path.split(file)[1])[0]
+
             if outlier_dict.get(image_name, (-1, None))[0] == 1:
                 logger.info(f'{self.name}: skipping outlier image in bg calculation')
                 continue
+
             bg_per_well_dict[well].append(bg_segment)
+            bg_fractions_per_well[well].append(bg_fractions_per_image[image_name])
+
         bg_per_well_dict = {well: torch.cat(bg_segments, dim=1) for well, bg_segments in bg_per_well_dict.items()}
         bg_per_well_stats = {well: self.get_bg_stats(bg_per_well_dict.get(well, None)) for well in wells}
+        bg_fractions_per_well = {well: np.median(bg_fractions_per_well[well]) if well in bg_fractions_per_well
+                                 else None for well in wells}
 
         bg_plate_stats = self.get_bg_stats(torch.cat(list(bg_per_well_dict.values()), dim=1)
                                            if len(bg_per_well_dict) > 0 else None)
+        plate_bg_fraction = [frac for frac in bg_fractions_per_well.values() if frac is not None]
+        plate_bg_fraction = np.median(plate_bg_fraction) if len(plate_bg_fraction) > 0 else None
```

Comment:
I think taking the median is totally fine to get an idea of the amount of BG in the plate.
But to really get the fraction of BG pixels in non-outlier images in the entire plate, somethig like 
`plate_bg_fraction = np.mean(np.concatenate([bg_fractions for bg_fractions in bg_fractions_per_well.values() if bg_fractions is not None]))`
should be correct.

Reply:
Yes, I agree. mean is more appropriate.